src/status_im/extensions/registry.cljs

```diff
@@ -0,0 +1,108 @@
+    (fx/merge cofx
+              (accounts.update/account-update {:extensions new-extensions} {:success-event nil})
+              (update-hooks extension-key))))
+
+(fx/defn install
+  [{:keys [db random-id-generator] :as cofx} extension-data]
+  (let [{:extensions/keys [manage]
+         :account/keys    [account]} db
+        {:keys [id url]} manage
+        extension      {:id      (-> (:value id)
+                                     (or (random-id-generator))
+                                     (string/replace "-" ""))
+                        :name    (get-in extension-data ['meta :name])
+                        :url     (:value url)
+                        :active? true}
+        new-extensions (assoc (:extensions account) (:url extension) extension)]
+    (fx/merge cofx
+              {:ui/show-confirmation {:title     (i18n/label :t/success)
+                                      :content   (i18n/label :t/extension-installed)
+                                      :on-accept #(re-frame/dispatch [:navigate-to-clean :my-profile])
+                                      :on-cancel nil}}
+              (accounts.update/account-update {:extensions new-extensions} {})
+              (add-to-registry (:value url) extension-data true))))
+
+(fx/defn uninstall
+  [{:keys [db] :as cofx} extension-key]
+  (let [{:account/keys [account]} db
+        new-extensions (dissoc (:extensions account) extension-key)]
+    (fx/merge cofx
+              {:ui/show-confirmation {:title     (i18n/label :t/success)
+                                      :content   (i18n/label :t/extension-uninstalled)
+                                      :on-accept nil
+                                      :on-cancel nil}}
+              (remove-from-registry extension-key)
+              (accounts.update/account-update {:extensions new-extensions} {}))))
+
+(fx/defn load
+  [cofx url]
+  (if (get-in cofx [:db :account/account :extensions url])
+    {:utils/show-popup {:title   (i18n/label :t/error)
+                        :content (i18n/label :t/extension-is-already-added)}}
+    {:extensions/load {:extensions [{:url     (string/trim url)
+                                     :active? true}]
+                       :follow-up  :extensions/stage}}))
+
+(fx/defn initialize
+  [{{:account/keys [account]} :db}]
+  (let [{:keys [extensions dev-mode?]} account]
+    (when dev-mode?
+      {:extensions/load {:extensions (vals extensions)
+                         :follow-up  :extensions/add-to-registry}})))
+
+(defn existing-hooks
+  [cofx extension-data]
+  (let [added-hooks (reduce (fn [acc [_ {:keys [hooks] :as v}]]
+                              (->> (:commands hooks)
+                                   (keys)
+                                   (into acc)))
+                            #{}
+                            (get-in cofx [:db :account/account :extensions]))
+        hooks       (->> (get-in extension-data [:data :hooks :commands])
```

Comment:
`:commands` should not be hardcoded, all hook types must be considered.
But the id is unique per hook type, not globally.